src/Application/Messenger/Handler/SendRequestConsumer.php

```diff
@@ -56,27 +56,37 @@ public function __invoke(SendRequest $sendRequest): void
             $this->fail($sendRequest->id, "Validation failed: {$this->validator->getReason()}");
         }
 
-        $bodyRenderedHtml = $this->twig->render("{$sendRequest->templateKey}.html.twig", $sendRequest->params);
+        // Instantiate a new Swift Message object
+        $email = new \Swift_Message();
+
+        $pathToImages = __DIR__ . '/../../../../images/';
+
+        $images['header'] = $email->embed(\Swift_Image::fromPath($pathToImages. 'TBG.jpg'));
+        $images['footer'] = $email->embed(\Swift_Image::fromPath($pathToImages . 'CCh.png'));
```

Comment:
@NoelLH So I've got the embed image to work, however they're showing up as attachments in Gmail, and just going through the [docs ](https://swiftmailer.symfony.com/docs/messages.html) it mentions this:

`The embedded files are sent in the email as a special type of attachment that has a unique ID used to reference them within your HTML attributes. On mail clients that do not support embedded files they may appear as attachments.`

Not ideal :/ - I've tried replacing `$images['header']` with this: `$images['header'] = someImageURL` and it's showing as a broken image.

Any ideas?

Reply:
@tbg-jasonfung Gmail supports embedded images. This is decades-old tech, and we're not doing the data URL thing they dislike discussed here. https://stackoverflow.com/a/42014708/2803757

I checked on some examples from when I last did this in 2016 and they load the image fine in Gmail. It also detects that part as having been embedded and doesn't show it as an attachment.

So, something is wrong with the CID refs or paths you're using, or possibly the types / dispositions of the parts in the mail. Have a look at the source and what the parts are named / encoded as?

Reply:
> Not ideal :/ - I've tried replacing `$images['header']` with this: `$images['header'] = someImageURL` and it's showing as a broken image.

Actually I take that back, looks like using SF public links don't work, i'll try using S3

Reply:
@tbg-jasonfung let's not get bogged down in different remote image solutions. We don't need any.

One thing I've just noticed is that in the other case I passed the HTML as the "primary" body in `setBody()`, which makes more sense as only the HTML can reference embedded images. Try swapping how the HTML and text parts get set up?

Reply:
@NoelLH Yes! That did the trick! Thanks